spring-integration-mongodb/src/main/java/org/springframework/integration/mongodb/outbound/ReactiveMongoDbStoringMessageHandler.java

```diff
@@ -0,0 +1,129 @@
+ *
+ * @since 5.3
+ */
+public class ReactiveMongoDbStoringMessageHandler extends AbstractReactiveMessageHandler {
+
+	private volatile ReactiveMongoOperations mongoTemplate;
+
+	private volatile ReactiveMongoDatabaseFactory mongoDbFactory;
+
+	private volatile MongoConverter mongoConverter;
+
+	private volatile StandardEvaluationContext evaluationContext;
+
+	private volatile Expression collectionNameExpression = new LiteralExpression("data");
+
+	private volatile boolean initialized = false;
+
+	/**
+	 * Will construct this instance using provided {@link ReactiveMongoDatabaseFactory}
+	 *
+	 * @param mongoDbFactory The reactive mongoDatabase factory.
+	 */
+	public ReactiveMongoDbStoringMessageHandler(ReactiveMongoDatabaseFactory mongoDbFactory) {
+		Assert.notNull(mongoDbFactory, "'mongoDbFactory' must not be null");
+
+		this.mongoDbFactory = mongoDbFactory;
+	}
+
+	/**
+	 * Will construct this instance using fully created and initialized instance of provided
+	 * {@link ReactiveMongoOperations}
+	 *
+	 * @param mongoTemplate The ReactiveMongoOperations implementation.
+	 */
+	public ReactiveMongoDbStoringMessageHandler(ReactiveMongoOperations mongoTemplate) {
+		Assert.notNull(mongoTemplate, "'mongoTemplate' must not be null");
+
+		this.mongoTemplate = mongoTemplate;
+	}
+
+	/**
+	 * Allows you to provide custom {@link MongoConverter} used to assist in serialization of
+	 * data written to MongoDb. Only allowed if this instance was constructed with a
+	 * {@link MongoDbFactory}.
+	 *
+	 * @param mongoConverter The mongo converter.
+	 */
+	public void setMongoConverter(MongoConverter mongoConverter) {
+		Assert.isNull(this.mongoTemplate,
+				"'mongoConverter' can not be set when instance was constructed with MongoTemplate");
+		this.mongoConverter = mongoConverter;
+	}
+
+	/**
+	 * Sets the SpEL {@link Expression} that should resolve to a collection name used by
+	 * {@link MongoOperations} to store data
+	 *
+	 * @param collectionNameExpression The collection name expression.
+	 */
+	public void setCollectionNameExpression(Expression collectionNameExpression) {
+		Assert.notNull(collectionNameExpression, "'collectionNameExpression' must not be null");
+		this.collectionNameExpression = collectionNameExpression;
+	}
+
+	@Override
+	public String getComponentType() {
+		return "mongo:reactive-outbound-channel-adapter";
+	}
+
+	@Override
+	protected void onInit() {
+		this.evaluationContext = ExpressionUtils.createStandardEvaluationContext(this.getBeanFactory());
+		if (this.mongoTemplate == null) {
+			this.mongoTemplate = new ReactiveMongoTemplate(this.mongoDbFactory, this.mongoConverter);
+		}
+		this.initialized = true;
+	}
+
+	@Override
+	protected Mono<Void> handleMessageInternal(Message<?> message) {
+		Assert.isTrue(this.initialized, "This class is not yet initialized. Invoke its afterPropertiesSet() method");
+		String collectionName = this.collectionNameExpression.getValue(this.evaluationContext, message, String.class);
```

Comment:
Right, because we don't do "polishing" for the code just because we notices some inconsistency.
Whenever we touch the code in the class, we try to refactor its content according our current code style requirements.
Gary insist to do that in the separate commit (if do that at all), but I'm OK to review the whole change at once.

So, yeah, you can leave code style for me: I'll try to fix it during merge having all your logic on board already.